modules/packages/Crypto.chpl

```diff
@@ -638,6 +639,169 @@ module Crypto {
     }
   }
 
+pragma "no doc"
+proc bfEncrypt(plaintext: CryptoBuffer, key: CryptoBuffer, IV: CryptoBuffer, cipher: CONST_EVP_CIPHER_PTR) {
+
+    var ctx: EVP_CIPHER_CTX;
+    EVP_CIPHER_CTX_init(ctx);
+
+    var keyData = key.getBuffData();
+    var ivData = IV.getBuffData();
+    var plaintextData = plaintext.getBuffData();
+    var plaintextLen = plaintext.getBuffSize();
+
+    var ciphertextLen = plaintextLen + 8; // block size length for blowfish
+    var cipherDomain: domain(1) = {0..#ciphertextLen};
+    var ciphertext: [cipherDomain] uint(8);
+    var updateCipherLen: c_int;
+
+    EVP_EncryptInit_ex(ctx,
+                       cipher,
+                       c_nil: ENGINE_PTR,
+                       c_ptrTo(keyData): c_ptr(c_uchar),
+                       c_ptrTo(ivData): c_ptr(c_uchar));
+    EVP_EncryptUpdate(ctx,
+                      c_ptrTo(ciphertext): c_ptr(c_uchar),
+                      c_ptrTo(ciphertextLen): c_ptr(c_int),
+                      c_ptrTo(plaintextData): c_ptr(c_uchar),
+                      plaintextLen: c_int);
+    EVP_EncryptFinal_ex(ctx,
+                        c_ptrTo(ciphertext[ciphertextLen..]): c_ptr(c_uchar),
+                        c_ptrTo(updateCipherLen): c_ptr(c_int));
+    cipherDomain = {0..#(ciphertextLen + updateCipherLen)};
+    return ciphertext;
+  }
+
+  pragma "no doc"
+  proc bfDecrypt(ciphertext: CryptoBuffer, key: CryptoBuffer, IV: CryptoBuffer, cipher: CONST_EVP_CIPHER_PTR) {
+
+    var ctx: EVP_CIPHER_CTX;
+    EVP_CIPHER_CTX_init(ctx);
+
+    var keyData = key.getBuffData();
+    var ivData = IV.getBuffData();
+    var ciphertextData = ciphertext.getBuffData();
+    var ciphertextLen = ciphertext.getBuffSize();
+
+    var plaintextLen = ciphertextLen;
+    var updatedPlainLen: c_int = 0;
+    var plainDomain: domain(1) = {0..plaintextLen};
+    var plaintext: [plainDomain] uint(8);
+
+    EVP_DecryptInit_ex(ctx,
+                       cipher,
+                       c_nil: ENGINE_PTR,
+                       c_ptrTo(keyData): c_ptr(c_uchar),
+                       c_ptrTo(ivData): c_ptr(c_uchar));
+    EVP_DecryptUpdate(ctx,
+                      c_ptrTo(plaintext): c_ptr(c_uchar),
+                      c_ptrTo(plaintextLen): c_ptr(c_int),
+                      c_ptrTo(ciphertextData): c_ptr(c_uchar),
+                      ciphertextLen: c_int);
+    EVP_DecryptFinal_ex(ctx,
+                        c_ptrTo(plaintext[plaintextLen..]): c_ptr(c_uchar),
+                        c_ptrTo(updatedPlainLen): c_ptr(c_int));
+
+    plainDomain = {0..#(plaintextLen + updatedPlainLen)};
+    return plaintext;
+  }
+
+  class Blowfish {
+    pragma "no doc"
+    var cipher: CONST_EVP_CIPHER_PTR;
+    pragma "no doc"
+    var byteLen: int;
+
+    /* The `Blowfish` class constructor that initializes the Blowfish encryption
+       algorithm with the right key length and chaining mode.
+
```

Comment:
It would be nice to list the valid modes in this comment along with what they mean. Even just spelling out what the acronyms stand for would be good.